realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -1668,4 +1670,107 @@ public void removeChangeListeners_throwOnUnmanagedObject() {
             looperThread.testComplete();
         }
     }
+
+    // Bug https://github.com/realm/realm-java/issues/2569
+    @Test
+    @RunTestInLooperThread
+    public void addChangeListener_returnedObjectOfCopyToRealmOrUpdate() {
+        looperThread.realm.beginTransaction();
```

Comment:
In other tests we use `Realm realm = looperThread.realm;` it makes the test easier to read.